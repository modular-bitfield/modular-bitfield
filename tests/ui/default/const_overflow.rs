use modular_bitfield::prelude::*;

#[bitfield]
pub struct OverflowTest {
    #[default = 256]  // Too large for B8 (max 255)
    overflow_b8: B8,
}

fn main() {}