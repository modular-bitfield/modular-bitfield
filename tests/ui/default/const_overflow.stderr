error: literal out of range for `u8`
 --> tests/ui/default/const_overflow.rs:5:17
  |
5 |     #[default = 256]  // Too large for B8 (max 255)
  |                 ^^^
  |
  = note: the literal `256` does not fit into the type `u8` whose range is `0..=255`
  = note: `#[deny(overflowing_literals)]` on by default
